add proper tests for torch.testing.make_tensor

pmeier opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #96334
* #96333
* #96124
* #97010
* #96332
* #97017
* #96125
* __->__ #96331

We had some minimal tests for `torch.testing.make_tensor` before, but nothing exhaustive. This lead to quite few edge cases being undetected. This PR adds comprehensive tests and leaves a few FIXMEs in there for behavior that needs to be fixed in `make_tensor`. This will happen in later commits of this stack. Meaning, at the end of this stack, there shouldn't be any FIXME left in the tests added here.

cc @mruberry

pmeier committed:
add proper tests for torch.testing.make_tensor

[ghstack-poisoned]

pmeier commented on test/test_testing.py:
```diff
@@ -1353,6 +1319,8 @@ def test_mismatching_values_msg(self):
                 fn()
 
 
+
+
```
Oops. Will revert.

mruberry commented on test/test_testing.py:
```diff
@@ -1398,6 +1366,180 @@ def test_matching_per_channel(self):
+        if dtype.is_complex:
+            t = torch.view_as_real(t)
+
+        self.assertTrue(((t >= low_inclusive) & (t < high_exclusive)).all())
```
```
_________ TestMakeTensorCUDA.test_low_high_default_smoke_cuda_bfloat16 _________
Traceback (most recent call last):
  File "/var/lib/jenkins/workspace/test/test_testing.py", line 1481, in test_low_high_default_smoke
    self.assertTrue(((t >= low_inclusive) & (t < high_exclusive)).all())
  File "/opt/conda/envs/py_3.10/lib/python3.10/unittest/case.py", line 687, in assertTrue
    raise self.failureException(msg)
AssertionError: tensor(False, device='cuda:0') is not true
```

The test failures look real -- maybe there's a bug with uniform for bfloat16 types?

pmeier committed:
Update on "add proper tests for torch.testing.make_tensor"


We had some minimal tests for `torch.testing.make_tensor` before, but nothing exhaustive. This lead to quite few edge cases being undetected. This PR adds comprehensive tests and leaves a few FIXMEs in there for behavior that needs to be fixed in `make_tensor`. This will happen in later commits of this stack. Meaning, at the end of this stack, there shouldn't be any FIXME left in the tests added here.

cc mruberry

[ghstack-poisoned]

pmeier committed:
Update on "add proper tests for torch.testing.make_tensor"


We had some minimal tests for `torch.testing.make_tensor` before, but nothing exhaustive. This lead to quite few edge cases being undetected. This PR adds comprehensive tests and leaves a few FIXMEs in there for behavior that needs to be fixed in `make_tensor`. This will happen in later commits of this stack. Meaning, at the end of this stack, there shouldn't be any FIXME left in the tests added here.

cc mruberry

[ghstack-poisoned]

pmeier committed:
Update on "add proper tests for torch.testing.make_tensor"


We had some minimal tests for `torch.testing.make_tensor` before, but nothing exhaustive. This lead to quite few edge cases being undetected. This PR adds comprehensive tests and leaves a few FIXMEs in there for behavior that needs to be fixed in `make_tensor`. This will happen in later commits of this stack. Meaning, at the end of this stack, there shouldn't be any FIXME left in the tests added here.

cc mruberry

[ghstack-poisoned]

mruberry reviewed (approved):
<img src="https://media1.giphy.com/media/RHIYhjyA2R8IibyqPU/giphy.gif"/>

pmeier committed:
Update on "add proper tests for torch.testing.make_tensor"


We had some minimal tests for `torch.testing.make_tensor` before, but nothing exhaustive. This lead to quite few edge cases being undetected. This PR adds comprehensive tests and leaves a few FIXMEs in there for behavior that needs to be fixed in `make_tensor`. This will happen in later commits of this stack. Meaning, at the end of this stack, there shouldn't be any FIXME left in the tests added here.

cc mruberry

[ghstack-poisoned]

pmeier committed:
Update on "add proper tests for torch.testing.make_tensor"


We had some minimal tests for `torch.testing.make_tensor` before, but nothing exhaustive. This lead to quite few edge cases being undetected. This PR adds comprehensive tests and leaves a few FIXMEs in there for behavior that needs to be fixed in `make_tensor`. This will happen in later commits of this stack. Meaning, at the end of this stack, there shouldn't be any FIXME left in the tests added here.

cc mruberry

[ghstack-poisoned]

pmeier committed:
Update on "add proper tests for torch.testing.make_tensor"


We had some minimal tests for `torch.testing.make_tensor` before, but nothing exhaustive. This lead to quite few edge cases being undetected. This PR adds comprehensive tests and leaves a few FIXMEs in there for behavior that needs to be fixed in `make_tensor`. This will happen in later commits of this stack. Meaning, at the end of this stack, there shouldn't be any FIXME left in the tests added here.

cc mruberry

[ghstack-poisoned]

pmeier committed:
Update on "add proper tests for torch.testing.make_tensor"


We had some minimal tests for `torch.testing.make_tensor` before, but nothing exhaustive. This lead to quite few edge cases being undetected. This PR adds comprehensive tests and leaves a few FIXMEs in there for behavior that needs to be fixed in `make_tensor`. This will happen in later commits of this stack. Meaning, at the end of this stack, there shouldn't be any FIXME left in the tests added here.

cc mruberry

[ghstack-poisoned]
